Unix/tools/allocdump

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+SCRIPTDIR=$(dirname "$0")
+BINDIR=/opt/omi/bin
+OMISERVER=${BINDIR}/omiserver
+GDB=gdb
+GDB_SCRIPT="$SCRIPTDIR/gdb_allocdump"
+MINIMUM_VERSION="100401000"   # version 1.4.1-0
+CONFIGEDITOR=${BINDIR}/omiconfigeditor
+CONF_FILE=/etc/opt/omi/conf/omiserver.conf
+
+# Verify that the GDB script file exists
+if [ ! -f ${GDB_SCRIPT} ]; then
+    echo "Gdb script file ${GDB_SCRIPT} does not exist"
+    exit 1
+fi
+
+# Verify that gdb exists
+if [ ! -x "$(command -v ${GDB})" ]; then 
+    echo "${GDB} not found in your PATH"
+    exit 1
+fi
+
+# Verify sudo
+EFFECTIVE_ID="$(id -u)"
+SUDO=
+if [ ${EFFECTIVE_ID} -ne 0 ]; then
+    SUDO=sudo
+    if [ ! -x "$(command -v ${SUDO})" ]; then 
+        echo "${SUDO} not found in your PATH"
+        exit 1
+    fi
+    ${SUDO} -v >&1 >/dev/null 
+    if [ $? -eq 1 ]; then
```

Comment:
Hmm. This is fine for a "blanket" `/etc/sudoers` file (i.e. you have perms to do anything). But what about restricted commands? If `/etc/sudoers` is restricted to certain commands, you can update your credentials, but that doesn't mean you can actually run this script.

Not sure how important this is. I guess, if you catch the error and print something meaningful, this is fine.

Reply:
No change.

Reply:
How about changing line 34 to read:

```
if [ $? -ne 0 ]; then
```

Seems safer ...